### ISSUE TITLE
Please add minimized-member-names=false to all template projects, and getting started guide

### ISSUE THREAD

**Issue by reubenfirmin** [rjaros/kvision#416]
Without the following, the IR compiler minifies member names. The minification doesn't produce stable names, so subsequent recompiles may produce different names. This means that serialization of data classes to localstorage isn't reliable.

```
    js(IR) {
        compilations.all {
            compileKotlinTask.kotlinOptions.freeCompilerArgs += listOf("-Xir-minimized-member-names=false")
        }
   }
```

**Comment by reubenfirmin** [comment]
...actually though, this doesn't seem to work, since I'm observing names change again even with this set. Debugging further.

**Comment by reubenfirmin** [comment]
Examples of different generations of the same object type attached.

![2022-07-03_19-17](https://user-images.githubusercontent.com/14217993/177065672-c1cadb28-be86-4c29-a1f6-e7b1a53549c3.png)

**Comment by reubenfirmin** [comment]
This didn't actually help. I had to implement my own serialization/deserialization logic. I filed bugs on kotlinjs instead; still, in lieu of them doing anything it might be nice to provide utility functions to do this.

**Comment by rjaros** [comment]
This is how IR backend is designed. You should just to use `kotlinx.serialization` for this.

**Comment by reubenfirmin** [comment]
Yeah I saw that here. https://kotlinlang.org/docs/js-ir-compiler.html#minification-of-member-names-in-production is where they suggest the above compile options, but they don't seem to work.

I'll try kotlinx.serialization next; I had rolled my own json serialization which works well enough for my needs after finding this issue, but if kotlinx.serialization doesn't add too much to my dist size it might be more convenient.
